src/containerapp/azext_containerapp/custom.py

```diff
@@ -854,12 +854,12 @@ def containerapp_up(cmd,
             _get_registry_from_app(app, source)  # if the app exists, get the registry
         _get_registry_details(cmd, app, source)  # fetch ACR creds from arguments registry arguments
 
-    app.create_acr_if_needed()
-
     if source:
-        app.run_acr_build(dockerfile, source, quiet=False, build_from_source=not _has_dockerfile(source, dockerfile))
+        used_default_container_registry = app.run_source_to_cloud_flow(source, dockerfile)
+    else:
+        app.create_acr_if_needed()
 
-    app.create(no_registry=bool(repo))
+    app.create(no_registry=bool(repo or used_default_container_registry))
```

Comment:
@daniv-msft Not necessarily related to these changes, but @Greedygre curious if you know why it is a no_registry scenario if `repo` was provided? Don't we need to pass the registry details to the GitHub action for non-cloud build scenarios?